sources/packages/api/src/route-controllers/designation-agreement/designation-agreement.controller.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  ForbiddenException,
+  Get,
+  NotFoundException,
+  Param,
+  Post,
+} from "@nestjs/common";
+import { IInstitutionUserToken } from "../../auth/userToken.interface";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import {
+  AllowAuthorizedParty,
+  IsInstitutionAdmin,
+  UserToken,
+} from "../../auth/decorators";
+import { DesignationAgreementService, FormService } from "../../services";
+import { getUTCNow } from "../../utilities";
+import {
+  GetDesignationAgreementDto,
+  SubmitDesignationAgreementDto,
+} from "./models/designation-agreement.model";
+import { InstitutionUserRoles } from "../../auth/user-types.enum";
+import { FormNames } from "../../services/form/constants";
+
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@Controller("institution/designation-agreement")
+export class DesignationAgreementController {
+  constructor(
+    private readonly designationAgreementService: DesignationAgreementService,
+    private readonly formService: FormService,
+  ) {}
+
+  /**
+   * Initiates a new designation agreement request. This action
+   * is meant to me initiated by the institution signing officer
+   * for further assessment of the Ministry.
+   * @param institutionId institution id requesting the designation.
```

Comment:
There is no param institutionId here. Instead payload is there.

Reply:
I will remove it 😉